### PR TITLE
Provide .editorconfig support

### PR DESCRIPTION
**Pull request by angelozerr**
This PR provides .editorconfig support without editing of .editorconfig file (without syntax coloration, without completion, hover, validation, codemining, etc), in other words it provide te capability to write an .editorconfig file to customize tabs, spaces, charset, eol, etc for all files opened with generic editor.

### CODE REVIEW

File: org.eclipse.editorconfig/plugin.properties [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+# are made available under the terms of the Eclipse Public License 2.0
+# which accompanies this distribution, and is available at
+# https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#     Angelo Zerr <angelo.zerr@gmail.com> - Initial API and implementation 
+###############################################################################
+pluginName=EditorConfig
```
Comment (mickaelistria) [inline org.eclipse.editorconfig/plugin.properties:14]:
```suggestion
pluginName=Support for editorconfig in Eclipse text editors
```

File: org.eclipse.editorconfig/plugin.xml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+         point="org.eclipse.ui.editors">
+      <editorContentTypeBinding
+            contentTypeId="org.eclipse.editorconfig"
+            editorId="org.eclipse.ui.genericeditor.GenericEditor">
+      </editorContentTypeBinding>
+   </extension>
+   
+   <extension
+	       point="org.eclipse.ui.startup">	
+	       <startup class="org.eclipse.editorconfig.internal.EditorConfigStartup" />
```
Comment (mickaelistria) [inline org.eclipse.editorconfig/plugin.xml:42]: I believe none of those are still necessary.

File: org.eclipse.editorconfig/src/org/eclipse/editorconfig/EditorConfigPlugin.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
+/**
+ *  Copyright (c) 2017 Angelo ZERR.
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  which accompanies this distribution, and is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
```
Comment (mickaelistria) [inline org.eclipse.editorconfig/src/org/eclipse/editorconfig/EditorConfigPlugin.java:6]: Needs to be EPLv2

File: org.eclipse.editorconfig/src/org/eclipse/editorconfig/internal/EditorConfigPreferenceStoreAdapter.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package org.eclipse.editorconfig.internal;
```
Comment (mickaelistria) [inline org.eclipse.editorconfig/src/org/eclipse/editorconfig/internal/EditorConfigPreferenceStoreAdapter.java:1]: Missing copyright header.

File: org.eclipse.editorconfig/src/org/eclipse/editorconfig/internal/EditorTracker.java [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+ * 
+ * <ul>
+ * <li>update {@link IPreferenceStore} of the opened text editor by adding the
+ * {@link EditorConfigPreferenceStore}.</li>
+ * <li>call {@link EditorConfigPreferenceStore#applyConfig()} when editor has
+ * focus to apply properties coming from .editorconfig files.</li>
+ * </ul>
+ *
+ */
+public class EditorTracker implements IWindowListener, IPageListener, IPartListener {
```
Comment (mickaelistria) [inline org.eclipse.editorconfig/src/org/eclipse/editorconfig/internal/EditorTracker.java:43]: Is this still necessary as the preferencestore is now provided as adapter?

File: org.eclipse.ui.genericeditor/src/org/eclipse/ui/internal/genericeditor/preferences/DocumentPreferenceStore.java [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+	public int getInt(String name) {
+		if (delegate != null) {
+			return delegate.getInt(name);
+		}
+		return 0;
+	}
+
+	@Override
+	public long getLong(String arg0) {
+		// TODO Auto-generated method stub
```
Comment (mickaelistria) [inline org.eclipse.ui.genericeditor/src/org/eclipse/ui/internal/genericeditor/preferences/DocumentPreferenceStore.java:112]: We'll need all those todos to be fixed.

**Review by mickaelistria — CHANGES_REQUESTED** [review]

File: org.eclipse.editorconfig/META-INF/MANIFEST.MF [hunk 7]
```diff
@@ -0,0 +1,30 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: %pluginName
+Bundle-Vendor: %providerName
+Bundle-Localization: plugin
+Bundle-SymbolicName: org.eclipse.editorconfig;singleton:=true
+Bundle-Version: 0.1.0.qualifier
```
[the remark this answers is not in the document]
Comment (angelozerr) [reply org.eclipse.editorconfig/META-INF/MANIFEST.MF]: fixed

File: org.eclipse.editorconfig/META-INF/MANIFEST.MF [hunk 8]
```diff
@@ -0,0 +1,30 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: %pluginName
+Bundle-Vendor: %providerName
+Bundle-Localization: plugin
+Bundle-SymbolicName: org.eclipse.editorconfig;singleton:=true
```
[the remark this answers is not in the document]
Comment (angelozerr) [reply org.eclipse.editorconfig/META-INF/MANIFEST.MF]: fixed

**Comment by sebthom** [comment]
@angelozerr any specific reason why you added the ec4j jars to the git repo? would it not be better to reference the maven artifacts in the build target file and include them in the update site?

**Comment by angelozerr** [comment]
> @angelozerr any specific reason why you added the ec4j jars to the git repo? would it not be better to reference the maven artifacts in the build target file and include them in the update site?

Indeed I should do that but I did that very quickly. I'm very busy with other features, but I will work on this PR again as soon as I will find time.

**Comment by angelozerr** [comment]
Just for your information IJ uses ec4j https://github.com/search?q=repo%3AJetBrains%2Fintellij-community%20ec4j&type=code

**Pull request closed without merging**